meetings/2021-12-01.md

```diff
@@ -0,0 +1,124 @@
+* Henry: Yes, it would be important to know what has been found.
+
+
+### Authorization Server
+
+* https://github.com/solid/authorization-panel/discussions/271
+* https://solid.github.io/data-interoperability-panel/specification/#data-grant
+* https://github.com/solid/solid-oidc/pull/18#issuecomment-974897450
+
+* Elf: Traditionally in OAuth, the access token is responsible for authz. It should give access including the notion of scopes. We don't have that in Solid.
+* ...: What we have in interop shows there is an AS and AA. You can see the example of a data grant. It's a way that always has one grantee. This is a projection on the user. That's where the application starts discovery.
+* ...: [The data grant tells you what you can access](https://solid.github.io/data-interoperability-panel/specification/#data-grant).
+* ...: Interestingly, you can also delegate data grants. A delegated grant is equivalent to OAuth scopes.
+* ...: In the PR, you can see that the resource server doesn't have to understand the grant. The AS issues an opaque token. And using token introspection, you could get understanding of what can be accessed.
+* ...: The AS could for example write appropriate ACRs on the RS.
+* ...: The application refers to the original data grant and the AS can verify that Alice gave access to Bob there. It is mostly now about coordinating the details of this AS. In other words agree on which parts of the delegated data grant to include in the Authorization Token.
+* Matthieu: We need to agree which parts of data grant get included in claim token.
+* Elf: In the end the Resource server has to enforce that.
+* Matthieu: In my mind, the RS talks to the AS and if the AS says agent X has the correct access modes, it performs the entailed operations accordingly.
+* Elf: The source of truth would be the AS and it would incorporate the delegated data grant. Bob is using their authorization agent to set the authorization grant. What I say is that ACP is one way that could enforce such data grants. Who's actually enforcing that? The AS is deciding who can have access but in the end the RS serves the resource.
+* ...: In my opinion the client should not need to understand anything about ACP, just the data grants. The RS should not have to understand data grants because the AS has a way to coordinate with the RS. So in a way the AS would pretty much translate the data grants into ACRs. AS would need to understand both (ACP and data grants). The RS would have data grants and project them as ACRs.
+* Elf: One question here is how Alice is setting the access policies?
+* ...: The way Alice is granting access to Bob, it's Bob's choice what client they use. So Alice gives access to Bob and Bob gives access to a client (delegated data grants, narrowing access). So Bob can delegate read only access to a client using a data grant.
+* ...: Alice uses her authz server. I don't think Alice sets ACRs, Alice uses a UI. Bob uses his authorization agent.
```

Comment:
```suggestion
* Matthieu: In my mind, the RS talks to the AS, and if the AS says agent X has the correct access modes, it performs the entailed operations accordingly.
* Elf: The source of truth would be the AS, and it would incorporate the delegated data grant. Bob is using their authorization agent to set the authorization grant. What I say is that ACP is one way that such data grants could be enforced. Who's actually enforcing that? The AS is deciding who can have access but in the end the RS serves the resource.
  * In my opinion, the client should not need to understand anything about ACP; just the data grants. The RS should not have to understand data grants, because the AS has a way to coordinate with the RS. So in a way, the AS would pretty much translate the data grants into ACRs. AS would need to understand both (ACP and data grants). The RS would have data grants and project them as ACRs.
* Elf: One question here is, how Alice is setting the access policies?
  * The way Alice is granting access to Bob, it's Bob's choice what client they use. So Alice gives access to Bob, and Bob gives access to a client (delegated data grants, narrowing access). So Bob can delegate read-only access to a client using a data grant.
  * Alice uses her authz server. I don't think Alice sets ACRs; Alice uses a UI. Bob uses his authorization agent.
```